drivers/usb/device/usb_dc_stm32.c

```diff
@@ -987,20 +987,16 @@ void HAL_PCD_DataInStageCallback(PCD_HandleTypeDef *hpcd, u8_t epnum)
 void HAL_PCDEx_SetConnectionState(PCD_HandleTypeDef *hpcd, uint8_t state)
 {
 	struct device *usb_disconnect;
+	u8_t gpio_flags = DT_INST_0_ST_STM32_USB_DISCONNECT_GPIOS_FLAGS;
 
 	usb_disconnect = device_get_binding(
 				DT_INST_0_ST_STM32_USB_DISCONNECT_GPIOS_CONTROLLER);
+
 	gpio_pin_configure(usb_disconnect,
-			   DT_INST_0_ST_STM32_USB_DISCONNECT_GPIOS_PIN, GPIO_DIR_OUT);
+			   DT_INST_0_ST_STM32_USB_DISCONNECT_GPIOS_PIN,
+			   GPIO_OUTPUT | gpio_flags);
 
-	if (state) {
-		gpio_pin_write(usb_disconnect,
-			       DT_INST_0_ST_STM32_USB_DISCONNECT_GPIOS_PIN,
-			       DT_INST_0_ST_STM32_USB_DISCONNECT_GPIOS_FLAGS);
-	} else {
-		gpio_pin_write(usb_disconnect,
-			       DT_INST_0_ST_STM32_USB_DISCONNECT_GPIOS_PIN,
-			       !DT_INST_0_ST_STM32_USB_DISCONNECT_GPIOS_FLAGS);
-	}
+	gpio_pin_set(usb_disconnect,
```

Comment:
Should this instead use the `usb_vbus_set()` that is already handled [here](https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/usb/usb_device.c#L995)? cc @erwango

Reply:
@carlescufi USB disconnect pin just enables a pull-up on D+. I don't think it is related to USB VBUS.

Reply:
Very well, thanks @ydamigos